### PR TITLE
Hotfix `TransformArrayItem`

### PR DESCRIPTION
**Pull request by tjzel**
## Summary

It was working when I was testing it before, idk what happened now but this fixes it anyway.

### CODE REVIEW

**Review by Latropos — APPROVED** [review]
🔥🔨

File: package.json [hunk 1]
```diff
@@ -23,7 +23,11 @@
     "format:docs": "cd docs && yarn format && cd ..",
     "find-unused-code:js": "yarn ts-prune --ignore \"index|.web.\" --error  ",
     "type:check": "yarn tsc --noEmit && cd plugin && yarn type:check && cd ..",
-    "type:check-api": "yarn tsc --noEmit --target es6 --module ESNext --jsx react-native --skipLibCheck true --allowSyntheticDefaultImports true --moduleResolution node --esModuleInterop true --strict true --forceConsistentCasingInFileNames true --resolveJsonModule true app/src/App.tsx __typetests__/react-native-reanimated-tests.tsx",
+    "type:run-test": "yarn tsc --noEmit --target es6 --module ESNext --jsx react-native --skipLibCheck true --allowSyntheticDefaultImports true --moduleResolution node --esModuleInterop true --strict true --forceConsistentCasingInFileNames true --resolveJsonModule",
```
Comment (tomekzaw) [inline package.json:26]: What's the difference between `type:check` and `type:run-test`? The underlying commands look quite similar, can we also unify their names?

File: src/reanimated2/layoutReanimation/defaultTransitions/EntryExitTransition.ts [hunk 2]
```diff
@@ -175,10 +187,14 @@ export class EntryExitTransition
...
           const transformProp = objectKeys[0];
-          const current = (
-            value[transformProp as keyof TransformArrayItem] as AnimationObject
-          ).current;
+          const current = // TODO TYPESCRIPT
+            // @ts-ignore Read similar comment above.
+            // prettier-ignore
+            // (prettier loves to push this @ts-ignore out of here and break CI)
+            (value[transformProp as keyof TransformArrayItem] as AnimationObject)
+            .current;
```
Comment (tomekzaw) [inline src/reanimated2/layoutReanimation/defaultTransitions/EntryExitTransition.ts:197]: Can we refactor this code so we don't need `prettier-ignore` here?

File: .github/workflows/check-TS-react-native.yml [hunk 3]
```diff
@@ -28,7 +28,14 @@ jobs:
...
-      - name: Check types
-        run: yarn type:check
+      - name: Check source types
+        run: yarn type:check:src
+      - name: Check plugin types
+        run: yarn type:check:plugin
       - name: Check API
-        run: yarn type:check-api
+        run: yarn type:check:app
+      - name: Run common tests
```
Comment (tomekzaw) [inline .github/workflows/check-TS-react-native.yml:37]:
```suggestion
      - name: Run common typetests
```

File: .github/workflows/static-root-checks.yml [hunk 4]
```diff
@@ -28,7 +28,7 @@ jobs:
       - name: Check types
         run: yarn tsc --noEmit
       - name: Check api
```
Comment (tomekzaw) [inline .github/workflows/static-root-checks.yml:30]:
```suggestion
      - name: Check ???
```

**Review by tomekzaw — APPROVED** [review]

**Pull request merged**